src/applications/simple-forms/40-0247/components/FileField.jsx

```diff
@@ -0,0 +1,717 @@
+ * @typedef uiOptions
+ * @type {object}
+ * @property {string} buttonText='Upload' - upload button text
+ * @property {string} addAnotherLabel='Upload another' - upload another text,
+ *  replaces upload button text when greater than one upload is showing
+ * @property {string} ariaLabelAdditionalText='' - additional screen-reader text to be appended to upload button's aria-label attribute.
+ * @property {string} tryAgain='Try again' - button in enableShortWorkflow
+ * @property {string} newFile='Upload a new file' - button in enableShortWorkflow
+ * @property {string} cancel='Cancel' - button visible while uploading & in enableShortWorkflow
+ * @property {string} delete='Delete file' - delete button text
+ * @property {string} modalTitle='Are you sure you want to delete this file?' -
+ *  delete confirmation modal title
+ * @property {ModalContent} modalContent - delete confirmation modal content
+ * @property {string} yesButton='Yes, delete this' - modal Yes button text
+ * @property {string} noButton='No, keep this' - modal No button text
+ */
+/**
+ * FormData of supported files
+ * @typeof Files
+ * @type {object}
+ * @property {string} name - file name
+ * @property {boolean} uploading - flag indicating that an upload is in
+ *  progress
+ * @property {string} confirmationCode - uuid of uploaded file
+ * @property {string} attachmentId - form ID set by user
+ * @property {string} errorMessage - error message string returned from API
+ * @property {boolean} isEncrypted - (Encrypted PDF only; pre-upload only)
+ *  encrypted state of the file
+ * @property {DOMFileObject} file - (Encrypted PDF only) File object, used
+ *  when user submits password
+ */
+const FileField = props => {
+  const {
+    enableShortWorkflow,
+    errorSchema,
+    formContext,
+    formData = [],
+    idSchema,
+    onBlur,
+    onChange,
+    registry,
+    schema,
+    uiSchema,
+  } = props;
+
+  const files = formData || [];
+  const [progress, setProgress] = useState(0);
+  const [uploadRequest, setUploadRequest] = useState(null);
+  const [isUploading, setIsUploading] = useState(
+    files.some(file => file.uploading),
+  );
+  const [showRemoveModal, setShowRemoveModal] = useState(false);
+  const [removeIndex, setRemoveIndex] = useState(null);
+  const [initialized, setInitialized] = useState(false);
+
+  const previousValue = usePreviousValue(formData);
+  const fileInputRef = useRef(null);
+  const fileButtonRef = useRef(null);
+
+  const uiOptions = uiSchema?.['ui:options'];
+
+  const maxItems = schema.maxItems || Infinity;
+  const { SchemaField } = registry.fields;
+  const attachmentIdRequired = schema.additionalItems.required
+    ? schema.additionalItems.required.includes('attachmentId')
+    : false;
+  const uswds = uiOptions.uswds || null;
+
+  const content = {
+    upload: uiOptions.buttonText || 'Upload',
+    uploadAnother: uiOptions.addAnotherLabel || 'Upload another',
+    ariaLabelAdditionalText: uiOptions.ariaLabelAdditionalText || '',
```

Comment:
This new `ariaLabelAdditionalText` prop is the first of only two functional differences from Platform's original component.  [See file-diff screenshot in main description]

Reply:
maybe it would be good to put this comment in the code at the top of the file so someone looking at the file will immediately know what the differences

Reply:
File-description comment just added.  Thanks!